net/znc/files/znc.init

```diff
@@ -168,15 +170,21 @@ start() {
 		config_foreach add_listener listener
 		config_foreach add_user user
 
-		chown -hR ${RUNAS_USER:-nobody}:${RUNAS_GROUP:-nogroup} /tmp/etc/znc
+		chown -hR ${RUNAS_USER:-nobody}:${RUNAS_GROUP:-nogroup} ${ZNC_CONFIG_PATH}
 	fi
 
 	if [ "$EXTERNAL_CONFIG" -eq 1 -a "$RUNAS_USER"  ]
 	then
 		local SU=$(which su)
 		if [ "$SU" ]
 		then
-			$SU -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER
+			chown -hR ${RUNAS_USER:-nobody}:${RUNAS_GROUP:-nogroup} ${ZNC_CONFIG_PATH}
+			if [ "$RUNAS_SHELL" ]
+			then
+				$SU -s $RUNAS_SHELL -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER
+			else
+				$SU -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER
+			fi
```

Comment:
You can write this shorter as 
`$SU ${RUNAS_SHELL:+s $RUNAS_SHELL} -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER`